### PR TITLE
Prep for #2179. Abstract parsing frame header. Catch ASIO exceptions.

### PR DESCRIPTION
**Pull request by subtly**

### CODE REVIEW

**Comment by gavofyork** [comment]
doesn't build on lin/mac. has this been tested in the wild for 24h?

File: test/libp2p/net.cpp [hunk 1]
```diff
@@ -314,23 +314,23 @@ BOOST_AUTO_TEST_CASE(kademlia)
 	node.nodeTable->discover(); // ideally, joining with empty node table logs warning we can check for
 	node.setup();
 	node.populate();
-	clog << "NodeTable:\n" << *node.nodeTable.get() << endl;
+//	clog << "NodeTable:\n" << *node.nodeTable.get() << endl;
```
Comment (chriseth) [inline test/libp2p/net.cpp]: Remove?

**Comment by gavofyork** [comment]
rebase.

**Comment by subtly** [comment]
@gavofyork Rebased and all green. I ran this for several days on mac and windows.
@chriseth I'll save buildbot some cycles and remove comments next PR :)

**Comment by subtly** [comment]
Been running this a couple days now w/no problems.

File: libp2p/Session.cpp [hunk 2]
```diff
@@ -157,114 +156,25 @@ void Session::serviceNodesRequest()
...
-			}
-			break;
-		default:
+		// v4 frame headers are useless, offset packet type used
+		// v5 protocol type is in header, packet type not offset
+		if (_capId == 0 && _t < UserPacket)
+			return interpret(_t, _r);
+		if (m_info.protocolVersion >= 5)
+			for (auto const& i: m_capabilities)
+				if (_capId == (uint16_t)i.first.second)
```
Comment (chriseth) [inline libp2p/Session.cpp]: Please don't use c-style casts in combination with `auto` and pseudo-anonymous members like `first` and `second`.

Comment (subtly) [inline libp2p/Session.cpp]: Bit of a precedent already.
grep -A 1 -r -e 'for (auto const& ' *

Comment (subtly) [inline libp2p/Session.cpp]: I've cringed at the use of auto however in this case it's clear-cut as m_capabilities is the same throughout the codebase. The C style cast is there until the types on both sides get cleaned up. Note that this PR is prep.

**Pull request merged**